components/org.wso2.carbon.identity.keyrotation/src/main/java/org.wso2.carbon.identity.keyrotation/config/KeyRotationConfig.java

```diff
@@ -0,0 +1,424 @@
+    private static final KeyRotationConfig instance = new KeyRotationConfig();
+
+    public static KeyRotationConfig getInstance() {
+
+        return instance;
+    }
+
+    /**
+     * Load the configurations placed in the properties.yaml file.
+     *
+     * @return KeyRotation config object.
+     * @throws KeyRotationException Exception thrown if something unexpected happens during key rotation.
+     */
+    public KeyRotationConfig loadConfigs(String[] args) throws KeyRotationException {
+
+        String propertiesFilePath = Paths.get("components", "org.wso2.carbon.identity.keyrotation", "src",
+                "main", "resources", "properties.yaml").toString();
+        if (args.length > 0) {
+            propertiesFilePath = args[0];
+        }
+        File file = new File(propertiesFilePath);
+        if (!file.exists()) {
+            throw new KeyRotationException(
+                    "Error occurred, properties.yaml file not found in provided path, " + propertiesFilePath);
+        }
+        log.debug("Loading Key Rotation Configs from path: " + propertiesFilePath);
+        KeyRotationConfig keyRotationConfig =
+                KeyRotationServiceUtils.loadKeyRotationConfig(propertiesFilePath);
+        checkKeyRotationConfigs(keyRotationConfig);
+        log.debug("Successfully loaded the config file.");
+        return keyRotationConfig;
+    }
+
+    /**
+     * To check whether the loaded configurations are valid and not null.
+     *
+     * @param keyRotationConfig Configuration data needed to perform the task.
+     * @throws KeyRotationException Exception thrown if something unexpected happens during key rotation.
+     */
+    public void checkKeyRotationConfigs(KeyRotationConfig keyRotationConfig) throws KeyRotationException {
+
+        Field[] props = KeyRotationConfig.getInstance().getClass().getDeclaredFields();
+        try {
+            for (int i = 0; i < props.length - 2; i++) {
+                if (props[i].get(keyRotationConfig) == null) {
+                    throw new KeyRotationException(
+                            "Error occurred, null value found in property, " + props[i].getName());
+                }
+                if ("oldISHome".equals(props[i].getName()) || "newISHome".equals(props[i].getName())) {
+                    File file = new File(props[i].get(keyRotationConfig).toString());
+                    if (!file.exists()) {
+                        throw new KeyRotationException(
+                                "Error occurred while finding " + props[i].getName() + " path.");
+                    }
+                }
+            }
+        } catch (IllegalArgumentException | IllegalAccessException e) {
+            throw new KeyRotationException("Error occurred while checking for null values in the loaded properties " +
+                    "file, ", e);
+        }
+    }
+
+    /**
+     * Getter for the old secret key.
```

Comment:
Shall we change the method comment to "Get for the old secret key.". Change in other places as well

Reply:
Addressed with https://github.com/wso2/identity-tools/pull/13/commits/6b6184cfde572cff05cdeefedb6e9233fd11d0e8